openobserve: 0.14.7 -> 0.15.1

SandaruKasa opened the pull request:
Diff:·https://github.com/openobserve/openobserve/compare/v0.14.7...v0.15.1                                                  

Changelog:·https://github.com/openobserve/openobserve/releases/tag/v0.15.1 

Closes  #440826

Fixes (well, kinda; the build still fails, but this time during the tests) #429262

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

happysalada commented:
The diff looks good, would be nice to see if we can fix this completely.

happysalada commented:
Commit and pr title are wrong , if you can fix, it should be 14.7 -> 15.1 
, right now it just contains 15.1 twice

SandaruKasa committed:
openobserve: 0.14.7 -> 0.15.1

Diff: https://github.com/openobserve/openobserve/compare/v0.14.7...v0.15.1

Changelog: https://github.com/openobserve/openobserve/releases/tag/v0.15.1

brianmay commented:
I think I can't update this pull request. Will see if I can open a new pull request with just those two tests disabled.

But wouldn't be surprised to find that we still get errors in other tests.

SandaruKasa commented:
Tried building 0.15.3 and after 20 minutes on a 24 core box got this:
```
openobserve> failures:
openobserve> 
openobserve> ---- service::metadata::trace_list_index::tests::test_write stdout ----
openobserve> 
openobserve> thread 'service::metadata::trace_list_index::tests::test_write' panicked at src/service/metadata/trace_list_index.rs:181:14:
openobserve> called `Result::unwrap()` on an `Err` value: Message("Error getting schema: DbError# error error returned from database: (code: 1) no such table: meta performing operation on key /schema/default/metadata/trace_list_index")
openobserve> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
openobserve> 
openobserve> ---- service::logs::bulk::tests::test_ingest_basic_functionality stdout ----
openobserve> 
openobserve> thread 'service::logs::bulk::tests::test_ingest_basic_functionality' panicked at src/service/schema.rs:586:18:
openobserve> called `Result::unwrap()` on an `Err` value: Message("Error getting schema: DbError# error error returned from database: (code: 1) no such table: meta performing operation on key /schema/test-org/logs/test_stream")
openobserve> 
openobserve> 
openobserve> failures:
openobserve>     service::logs::bulk::tests::test_ingest_basic_functionality
openobserve>     service::metadata::trace_list_index::tests::test_write
openobserve> 
openobserve> test result: FAILED. 1744 passed; 2 failed; 5 ignored; 0 measured; 3 filtered out; finished in 1.06s
```

---


Closing in favor of #457421
